beacon-chain/db/state.go

```diff
@@ -117,10 +121,15 @@ func (db *BeaconDB) HeadState(ctx context.Context) (*pb.BeaconState, error) {
 
 		var err error
 		beaconState, err = createState(enc)
+		if err != nil {
+			return err
+		}
 		if beaconState != nil && beaconState.Slot > db.highestBlockSlot {
 			db.highestBlockSlot = beaconState.Slot
 		}
-		return err
+		copy(db.serializedState, enc)
```

Comment:
do you have to do `copy`? can it just be an assignment? `db.serializedState = enc`